### PR TITLE
New playbook to run database setup and update django local settings

### PR DESCRIPTION
**Pull request by rlskoeser**
- update postgres host variables for new servers
- update postgres version to 15
- new playbook to run postgres database setup tasks and update django local settings

## notes:
- specific tags are required for the new playbook to operate the way it's intended; I have it documented in the comments, but it bothers me a little
- I'm understanding ansible better, and I know now from trying it that this playbook doesn't work properly when run on multiple hosts at a time. I suspect this is because we have tasks using `register` where we should be using `set_fact`; this is something I'd be interested in improving

### CODE REVIEW

File: roles/python/tasks/install_dependencies.yml
Comment (rlskoeser) [inline roles/python/tasks/install_dependencies.yml:46]: this is an unrelated change: recently added logic to install python dependencies from a different config file (needed for simulating risk), but when running this discovered that an additional check is needed for when to run that install (geniza has the pyproject file to configure a tool but doesn't use it for python requirements)

File: roles/django/defaults/main.yml
Comment (rlskoeser) [inline roles/django/defaults/main.yml:37]: we have the same default path set in multiple places; the other places weren't getting loaded in this playbook. suggestions for a better solution?

File: group_vars/prosody/vars.yml
Comment (acozine) [inline group_vars/prosody/vars.yml:70]: Do you need the comment line now that the setting is gone?
```suggestion
```

File: roles/build_project_repo/tasks/main.yml
Comment (acozine) [inline roles/build_project_repo/tasks/main.yml:43]: Do you need to discover this on each machine? Could you define it in vars instead?

File: roles/postgresql/tasks/backup_db.yml
Comment (acozine) [inline roles/postgresql/tasks/backup_db.yml:4]: This needs to go under the `block` definition, if it applies to all tasks within the block.

**Review by acozine — COMMENTED** [review]
Here's what jumps out at me on my first pass. I'd be happy to pair on refining the role/playbook if you'd like - I haven't read the role or playbook in detail yet, though.

File: roles/django/defaults/main.yml
Comment (acozine) [reply roles/django/defaults/main.yml:37]: The role defaults file is a great place for default values that should apply across environments and servers. Just remember that any other value for that variable will override the role default.

File: roles/build_project_repo/tasks/main.yml
Comment (rlskoeser) [reply roles/build_project_repo/tasks/main.yml:43]: This is the version of the app that we're deploying, so it is specific to the app+install (which roughly equates to being specific to the machine, at least for our current setup). This is one of the things I was asking about when I joined ansible open house last week, it's a variable we use for paths but we have to run some code to determine it.

File: roles/postgresql/tasks/backup_db.yml
Comment (rlskoeser) [reply roles/postgresql/tasks/backup_db.yml:4]: I tried that and it doesn't work - I get a syntax error. I think it's behaving as expected?

However, I'm glad I took another look - when I tested this I did figure out that my recommended syntax for calling the playbook didn't need to specify the skip tags, since we're already limiting to specific tags.

**Comment by rlskoeser** [comment]
@acozine thank you for review and feedback, I've done some cleanup based on your suggestions and questions. Is there anything you think is important to address before merging?  

There are larger problems with cdh-ansible that I'd like to address at some point, and would love to get your input on priority & order once we do an overview/deep dive on current state, but don't want to hold this up.

File: roles/build_project_repo/tasks/main.yml
Comment (acozine) [reply roles/build_project_repo/tasks/main.yml:43]: Ah, so the variable you're establishing is not the python version, it's a deployment timestamp or something similar that changes with every fresh code deploy? If you're using a symlink  (like `current`) you might be able to define the correct path by specifying the symlink and `follow: true` instead of using this variable.

**Review by acozine — APPROVED** [review]
Looks good to merge. This makes the automation better, so let's not wait for more refinements. Happy to look at the variables issues together - maybe next week sometime?

File: roles/build_project_repo/tasks/main.yml
Comment (rlskoeser) [reply roles/build_project_repo/tasks/main.yml:43]: thank you for explaining that more clearly! yes, that's exactly the case

in fact, I remember now that I set a playbook-specific variable that uses the `current` symlink for this playbook, since we're not deploying a new version and the version that would be deployed isn't guaranteed to match the current version with the settings we want to update

**Comment by rlskoeser** [comment]
> Looks good to merge. This makes the automation better, so let's not wait for more refinements. Happy to look at the variables issues together - maybe next week sometime?

Thank you! That would be helpful, I've done some preliminary work in a different branch and I think I made some progress - I think that's when I tripped over / figured out the `set_fact`; it would be valuable to look at together.

**Pull request merged**
